CodePush.js

```diff
@@ -173,7 +173,7 @@ const notifyApplicationReady = (() => {
 async function notifyApplicationReadyInternal() {
   await NativeCodePush.notifyApplicationReady();
   const statusReport = await NativeCodePush.getNewStatusReport();
-  tryReportStatus(statusReport); // Don't wait for this to complete.
+  statusReport && tryReportStatus(statusReport); // Don't wait for this to complete.
```

Comment:
It seems to complicate it a fair bit, primarily due to [the retry logic](https://github.com/convoyinc/react-native-code-push/blob/2b628007e0298909e0063e5906cb890cd6ae40aa/CodePush.js#L209-L220)

Making that code behave the same, without checking the status report up front, is hard to follow; and it seems weird to only treat it as optional from `notifyApplicationReadyInternal` but not when recursing